src/core.h

```diff
@@ -17,40 +17,58 @@ class CVC;
 
 class Action {
   public:
+    Action(Character* actor, double score);
     //Get this action's actor (the character taking the action)
     Character* GetActor();
+    void SetActor(Character* actor);
+
     //Get the score this action's been given
     //it's assumed this is normalized against alternative actions
     //I'm not thrilled with this detail which makes it make sense ONLY in the
     //context of some other instances
     double GetScore();
+    void SetScore(double score);
 
     //Determine if this particular action is valid in the given gamestate by
     //the given character
-    virtual bool IsValid(CVC* gamestate) = 0;
+    virtual bool IsValid(const CVC& gamestate) = 0;
 
     //Have this action take effect by the given character
-    virtual void TakeEffect(CVC* gamestate) = 0;
+    virtual void TakeEffect(const CVC& gamestate) = 0;
 
   private:
     Character* actor_;
     double score_;
 };
 
+class TrivialAction : public Action {
+  public:
+    TrivialAction(Character *actor, double score);
+
+    //implementation of Action
+    bool IsValid(const CVC& gamestate);
+    void TakeEffect(const CVC& gamestate);
+
+};
+
 class DecisionEngine {
   public:
     //TODO: what's the right return type here? the idea is that this is a
     //factory of scored actions in a vector
+    std::vector<std::unique_ptr<Action>> EnumerateActions(const CVC& cvc, Character* character);
 };
 
 class CVC {
   public:
+    CVC(std::unique_ptr<DecisionEngine> decision_engine, std::vector<std::unique_ptr<Character>> characters);
```

Comment:
does this make sense? these are unique_ptrs which we're going to take ownership of, right?

Reply:
yeah, looks sane